File: img_load/src/main/java/com/ad/img_load/pickerview/pv/adapter/NumericWheelAdapter.java
package com.ad.img_load.pickerview.pv.adapter;


import com.ad.img_load.pickerview.wheelview.adapter.WheelAdapter;

/**
 * Numeric Wheel adapter.
 */
public class NumericWheelAdapter implements WheelAdapter {
	
	private final int minValue;
	private final int maxValue;

	/**
	 * Constructor
	 * @param minValue the wheel min value
	 * @param maxValue the wheel max value
	 */
	public NumericWheelAdapter(int minValue, int maxValue) {
		this.minValue = minValue;
		this.maxValue = maxValue;
	}

	@Override
	public Object getItem(int index) {
		if (index >= 0 && index < getItemsCount()) {
			int value = minValue + index;
			return value;
		}
		return 0;
	}

	@Override
	public int getItemsCount() {
		return maxValue - minValue + 1;
	}
	
	@Override
	public int indexOf(Object o){
		try {
			return (int)o - minValue;
		} catch (Exception e) {
			return -1;
		}
	}
}

File: Widgets/src/main/java/com/colin/library/android/widget/picker/adapter/NumericWheelAdapter.kt
package com.colin.library.android.widget.picker.adapter

import com.colin.library.android.utils.Constants

/**
 * Author:ColinLu
 * E-mail:945919945@qq.com
 * Create:2025-05-03 00:09
 *
 * Des   :NumericWheelAdapter
 */
class NumericWheelAdapter(private val minValue: Int, private val maxValue: Int) : WheelAdapter {

    override fun getItem(index: Int): Any {
        if (index >= 0 && index < getItemsCount()) {
            val value = minValue + index
            return value
        }
        return Constants.INVALID
    }

    override fun indexOf(data: Any): Int {
        return try {
            (data as Int) - minValue
        } catch (e: Exception) {
            -1
        }
    }

    override fun getItemsCount(): Int {
        return maxValue - minValue + 1
    }

}
